packages/react-router/lib/components.tsx

```diff
@@ -601,6 +602,105 @@ export function createRoutesFromChildren(
   return routes;
 }
 
+/**
+ * Converts Route objects with a `module` property that import a module that
+ * conforms to the Remix route module convention to React Router's standard
+ * route object. Properties directly set on the route object override exports
+ * from the route module.
+ */
+export function createModuleRoutes(
+  routes: (ModuleRouteObject | RouteObject)[]
+): RouteObject[] {
+  return routes.map((route) => {
+    if (!isModuleRouteObject(route)) {
+      return route;
+    }
+    const { module: moduleFactory, children, ...restOfRoute } = route;
+
+    let element;
+    if (!route.element) {
+      let Component = React.lazy(moduleFactory);
+      element = <Component />;
+    }
+
+    let loader: RouteObject['loader'] = route.loader;
+    if (typeof loader !== 'function') {
+      loader = async (args) => {
+        const mod = await moduleFactory();
+        return typeof mod.loader === 'function' ? mod.loader(args) : null;
+      };
+    }
+
+    let action: RouteObject['action'] = route.loader;
+    if (typeof action !== 'function') {
+      action = async (args) => {
+        const mod = await moduleFactory();
+        return typeof mod.action === 'function' ? mod.action(args) : null;
+      };
+    }
+
+    let errorElement = route.errorElement;
+    if (errorElement) {
```

Comment:
I think you want a `!errorElement` here
```suggestion
    if (!errorElement) {
```

Reply:
🤦 yup, thanks!